spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jdbc/DataSourceAutoConfiguration.java

```diff
@@ -73,6 +73,13 @@ public DataSourceInitializer dataSourceInitializer() {
 		return new DataSourceInitializer();
 	}
 
+	@Bean
+	@Conditional(PooledDataSourceCondition.class)
+	@ConditionalOnProperty(prefix = "spring.datasource", name = "early-pool-creation", havingValue = "true", matchIfMissing = true)
+	public PooledDataSourceBeanPostProcessor tomcatJdbcPooledDataSourceBeanPostProcessor() {
```

Comment:
Since this works for all DataSource implementations, the name of the method (and thus the bean) would be better without `tomcatJdbc` in it.